### DISCUSSION TITLE
Maybe Patch the 2017 Minecraft Staging Launcher

### DISCUSSION THREAD

**Discussion by Thundercom-Software** [olauncher/olauncher#74] [Ideas]
The following launcher is a staging version of the 2017 official Minecraft Launcher and after logging in to OLauncher it for some reason logs in to the staging launcher as well saying that my account is a Legacy Premium account not a Microsoft one. The launcher works with 1.12.2 as far as I'm aware but doesn't work with forge or optifine installs since it's in offline mode without being able to fetch server status. It would be nice if you could make this one functional again. I love it's interface and how simple it is to use and install. Thank you!

Windows: https://launcher.mojang.com/mc-staging/download/Minecraft_staging.zip
OSX: https://launcher.mojang.com/mc-staging/download/Minecraft_staging.dmg

The links above are from the official Reddit post back when they allowed people to test the launcher. Feel free to download them and scan them for viruses on VirusTotal if you wish.

**Comment by bigfoot547** [comment]
Do you know if it was written in java or C++? The methods used in olauncher only work because the original launcher was written in Java. C++ would be a bit more difficult.

**Reply by Thundercom-Software** [reply]
According to the information from the reddit post here: https://www.reddit.com/r/Minecraft/comments/59b7m0/help_us_test_the_new_minecraft_launcher_check_the/

"Do I need Java installed?
Nope. Not at all! We'll download and maintain our own version of Java, much like the current launcher does. This is used for the game, and can be overridden. The launcher itself doesn't need Java."

That's the information I could find for now. If you need more info let me know. Thank you!

**Reply by bigfoot547** [reply]
It's possible to pack Java applications into executables, but given context clues, I am pretty sure it's not written in Java.

**Reply by Thundercom-Software** [reply]
If it is possible to patch this launcher as well, I'd be more than happy to search for more information on it's development and if it's based on C++, Java, C# (possibly) or any other codebase. Let me know if plans to patch this launcher are set in the future. Thank you again!

**Reply by bigfoot547** [reply]
I currently don't plan to reverse engineer this launcher because it is probably more difficult than it's worth. I admire the staging launcher's simplicity and relative lightweightness, but would rather make a launcher from scratch to be similar to it (especially because it started to use the silly web stuff like the new launcher).

**Reply by Thundercom-Software** [reply]
Alright then. Looking forward to a somewhat remake of this launcher!

**Reply by RezyPlayz** [reply]
its made in c++

**Comment by DevBefell** [comment]
This is from [this post](https://www.reddit.com/r/Minecraft/comments/59b7m0/help_us_test_the_new_minecraft_launcher_check_the/), for reference there is also [this one](https://www.minecraftforum.net/news/60534-test-the-new-minecraft-launcher-today), I think it is Electron?  In the Linux one the launcher code has js and css as well as some Electron like stuff, could be wrong.

**Reply by bigfoot547** [reply]
I think they preferred a different web thingy than electron, but yeah it is likely something similar.
